### PR TITLE
host helm chart on github pages

### PR DESCRIPTION
**Pull request by gohclive**
setup github pages to host the helm chart.

### CODE REVIEW

**Comment by gohclive** [comment]
cc @rarkins for review

File: index.yaml
Comment (rarkins) [inline index.yaml:16]: Is this a typo? I don't know where 2.1.1 or 3.1.1 come from, but I'm guessing 2.1.1 is old and 3.1.1 is a typo?

**Comment by rarkins** [comment]
Which URL would this appear in if published?

File: index.yaml
Comment (gohclive) [reply index.yaml:16]: the version here is based on Chart.yaml version which is currently 3.1.2

**Review by rarkins — CHANGES_REQUESTED** [review]
I'm still uncertain what the result will be after this is merged. For example will it create and maintain the `gh-pages` branch? What will happen if the github.com/whitesource org doesn't have any existing github pages today?

File: .github/workflows/main.yml
Comment (gohclive) [reply .github/workflows/main.yml:11]: a gh-pages branch needs to be created first for the workflow to work. After that, github would automatically create a pages workflow to maintain the gh-pages whenever there's an update on main.

for the relase chart workflow , currently it is triggered manually when needed

Comment (gohclive) [reply .github/workflows/main.yml]: Pinned release

File: .github/workflows/main.yml
Comment (gohclive) [inline .github/workflows/main.yml:11]: pinned released

**Comment by gohclive** [comment]
Action required by maintainer after merge
1. create a gh-pages branch
2. add github token with repo, workflow & gist access into repo secrets named "GH_TOKEN"
3. trigger workflow action manually

File: .github/workflows/main.yml
Comment (onedr0p) [inline .github/workflows/main.yml:11]: Would it make more sense to use the official action from Helm?

https://github.com/helm/chart-releaser-action

Comment (gohclive) [reply .github/workflows/main.yml:11]: That would work as well, I was looking at #183 and decided to contribute based on it.

Comment (gohclive) [reply .github/workflows/main.yml]: i've updated the file to use the official action from the Helm team

**Comment by k-jingyang** [comment]
Hey @rarkins, what would make it easier for this to be merged?

**Comment by ammmze** [comment]
> I'm still uncertain what the result will be after this is merged. For example will it create and maintain the `gh-pages` branch? What will happen if the github.com/whitesource org doesn't have any existing github pages today?

@rarkins We would need you (or another maintainer) to perform steps 2 and 3 from the [helm releaser pre-requisites](https://github.com/helm/chart-releaser-action#pre-requisites). Once that is complete, the actions workflow will maintain the branch from there whenever helm chart version changes.

The `gh-pages` branch should be an orphaned branch, meaning it won't carry with it all the files and history from the `main` branch. You can create the `gh-pages` with the following:
```
git checkout --orphan gh-pages
git reset --hard
git commit --allow-empty -m "Initializing gh-pages branch"
git push origin gh-pages
git checkout main
```

**Comment by ammmze** [comment]
> Which URL would this appear in if published?

By default the url would be `https://{org}.github.io/{repo}`. So in this case that would be `https://whitesource.github.io/renovate-on-prem`. However, the helm releaser action doesn't actually create an `index.html` file, so while `https://whitesource.github.io/renovate-on-prem` would be the helm repo url, the actual file the helm releaser is maintaining would be at `https://whitesource.github.io/renovate-on-prem/index.yaml`. I've got an example of what that looks like [here](https://ammmze.github.io/crunchydata-postgres-operator-charts/index.yaml). This is maintained by this same helm releaser action.

File: .github/workflows/main.yml
Comment (ammmze) [inline .github/workflows/main.yml:27]: Should we update the README with instructions to use it? Something like this:

## Helm

### Add Helm repository

```shell
helm repo add renovate-on-prem https://whitesource.github.io/renovate-on-prem
helm repo update
```

### Install Renovate chart

```shell
helm install --generate-name --set renovate.config='\{\"token\":\"...\"\}' renovate-on-prem/whitesource-renovate
```

> See the available [values](./helm-charts/whitesource-renvoate/values.yaml) for full configuration.

**Comment by rarkins** [comment]
Thanks all. I'm going to move this repository into our new company name Mend (github.com/mend) first so that the URL won't be changed on everyone later.

**Review by rarkins — APPROVED** [review]

File: .github/workflows/main.yml
Comment (ammmze) [reply .github/workflows/main.yml:27]: Yep! Added [here](https://github.com/mend/renovate-on-prem/pull/234).

**Pull request merged**
